timeslots/repository.go

```diff
@@ -15,47 +18,88 @@ func NewRepository(db *sql.DB) *Repository {
 }
 
 func createTimeslotsTable() {
-	const create = `CREATE TABLE IF NOT EXISTS timeslots(id TEXT, ownerId TEXT, bookedById TEXT, timeFrom INTEGER, timeTo INTEGER, booked INTEGER)`
+	const create = `CREATE TABLE IF NOT EXISTS timeslots(id TEXT, ownerId TEXT, bookedById TEXT, timeFrom INTEGER, timeTo INTEGER, bookingDay INTEGER, booked INTEGER)`
 
 	if _, err := db.DbInstance.Exec(create); err != nil {
 		log.Fatal(err)
 	}
 }
 
-func (r Repository) createTimeslot(timeslot Timeslot) error {
+func insertTimeslotIntoDB(timeslot TimeslotData) error {
+	id := uuid.NewString()
+	bookingDayInUnix := timeslot.BookingDay.Unix()
+
+	for i := 0; i < len(timeslot.Time); i++ {
+		timeInUnixFrom := timeslot.Time[i].From.Unix()
+		timeInUnixTo := timeslot.Time[i].To.Unix()
+
+		t := TimeslotInDB{TimeslotBase: TimeslotBase{Id: id, OwnerId: timeslot.OwnerId, Booked: false}, TimeFrom: timeInUnixFrom, TimeTo: timeInUnixTo, BookingDay: bookingDayInUnix}
+
+		query := "INSERT INTO timeslots(id, ownerId, bookedById, timeFrom, timeTo, bookingDay, booked) values(?,?,?,?,?,?,?)"
+
+		_, err := db.DbInstance.Exec(query, t.Id, t.OwnerId, t.BookedById, t.TimeFrom, t.TimeTo, t.BookingDay, t.Booked)
+
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r Repository) createTimeslots(timeslot TimeslotData) error {
 	createTimeslotsTable()
 
-	// for i := 0; i < len(timeslots); i++ {
-	// add id, TimeFrom, TimeTo here
-	//}
+	bookingDayInUnix := timeslot.BookingDay.Unix()
 
-	// for _, timeslot := range timeslots {
-	// somehow construct SQL query here
-	// }
+	ts, err := r.getTimeslotByBookingDay(bookingDayInUnix)
 
-	timeInUnixFrom := timeslot.TimeFrom.Unix()
-	timeInUnixTo := timeslot.TimeTo.Unix()
+	if !errors.Is(err, sql.ErrNoRows) {
```

Comment:
instead of `sql.ErrNoRows` sql request count, if count > 0 -> there is a timeslot -> delete it